osu.Game/Scoring/Legacy/LegacyScoreEncoder.cs

```diff
@@ -143,7 +144,7 @@ private LegacyReplayFrame getLegacyFrame(ReplayFrame replayFrame)
                     return legacyFrame;
 
                 case IConvertibleReplayFrame convertibleFrame:
-                    return convertibleFrame.ToLegacy(beatmap);
+                    return convertibleFrame.ToLegacy(beatmap.AsNonNull());
```

Comment:
Please `Debug.Assert()` this instead.

Reply:
```
case IConvertibleReplayFrame convertibleFrame:
    Debug.Assert(beatmap != null);
    return convertibleFrame.ToLegacy(beatmap);
```
Something likes this?

But doesn't `AsNonNull()` already contains `Debug.Assert` inside?

Reply:
Yep. I think the general consensus is still to use Debug.Assert where code isn’t just working around NRT shortcomings, since it’s more explicit.